### DISCUSSION TITLE
Multi domain psrp issue with gmail

### DISCUSSION THREAD

**Discussion by marketfin** [postalserver/postal#2282] [Help with using Postal]
Hello,

We have setup multi vps with 6 different sub domains (smtp1.domain1.com, smtp2.domain1.com, smtp3.domain1.com, etc). Each server has 3 organizations and 3 domains setup as well (domain1.com, domain2.com, domain3.com).
 
We have setup psrp1, psrp2, psrp3, etc for each server for return path prefix. 
custom_return_path_prefix: psrp1

We have setup cname to point to correct rp.postal.domain

ISSUE:
Now issue comes with gmail. Once email is sent it throws error:

550-5.7.1 [ip 12] Our system has detected that this message is 550-5.7.1 likely unsolicited mail. To reduce the amount of spam sent to Gmail, 550-5.7.1 this message has been blocked. Please visit 550-5.7.1 https://support.google.com/mail/?p=UnsolicitedMessageError 550 5.7.1 for more information. t71-20020a63814a000000b00478ae301535si7472806pgd.532 - gsmtp

When we remove the psrp cname and the rp.postal.domain emails go out just fine. The only issue with this is since we have multiple domains going from single domain the spf alignment fails. Everything else is a PASS. 

SPF domain does not align with RFC5322.From domain (domain1.com != domain2.com). Alignment mode: relaxed.

SOLUTION?
We are thinking maybe gmail doesn't like cname and if we can somehow make postal get the return psrp to with TXT instead of CNAME this issue might go away. Not sure if this will work but might be worth a try. Is is possible to change to TXT?

No we are not blacklisted anywhere and yes reputation in google is bad.

**Comment by willpower232** [comment]
If you want to specify the TXT and MX records for rp manually instead of using a CNAME then that should still work but it might be worth having a look at the rest of the records incase the CNAME is including additional records that you didn't expect which is confusing Google somehow.

Naturally Postal won't be able to verify the records any more but that shouldn't be an issue, as long as the records exist the email should still be successful.
